### PR TITLE
8286825: Linker naming cleanup

### PR DESCRIPTION
**Pull request by JornVernee**
This patch is a batch naming cleanup for the foreign linker implementation.

The naming changes are as follows:

  - ProgrammableInvoker -> DowncallLinker
  - ProgrammableUpcallHandler -> UpcallLinker
  - 'native invoker' -> 'downcall stub'
  - 'optimzed entry blob' -> 'upcall stub'
  - OptimizedEntryBlob -> UpcallStub
  - optimized_entry_frame -> upcall_stub_frame

Then renaming of some hotspot files:

  - universalNativeInvoker* -> downcallLinker*
  - universalUpcallHandler* -> upcallLinker*
  - foreign_globals* -> foreignGlobals* (to match existing convention)
  
Method, field, and other variable names are also adjusted accordingly.

Testing: Tier 1-4

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286825](https://bugs.openjdk.java.net/browse/JDK-8286825): Linker naming cleanup

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - **Reviewer**)
 * [Robbin Ehn](https://openjdk.java.net/census#rehn) (@robehn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8777/head:pull/8777` \
`$ git checkout pull/8777`

Update a local copy of the PR: \
`$ git checkout pull/8777` \
`$ git pull https://git.openjdk.java.net/jdk pull/8777/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8777`

View PR using the GUI difftool: \
`$ git pr show -t 8777`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8777.diff">https://git.openjdk.java.net/jdk/pull/8777.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/riscv/upcallLinker_riscv.cpp [hunk 1]
```diff
@@ -27,12 +27,12 @@
...
-                                                                  bool needs_return_buffer, int ret_buf_size) {
+address UpcallLinker::generate_optimized_upcall_stub(jobject receiver, Method* entry,
+                                                     BasicType* in_sig_bt, int total_in_args,
+                                                     BasicType* out_sig_bt, int total_out_args,
+                                                     BasicType ret_type,
+                                                     jobject jabi, jobject jconv,
+                                                     bool needs_return_buffer, int ret_buf_size) {
   ShouldNotCallThis();
   return nullptr;
 }
```
Comment (mcimadamore) [inline src/hotspot/cpu/riscv/upcallLinker_riscv.cpp:38]: Missing newline (this was here before)

File: src/hotspot/cpu/riscv/upcallLinker_riscv.cpp [hunk 2]
```diff
@@ -27,12 +27,12 @@
 #include "prims/universalUpcallHandler.hpp"
 #include "utilities/debug.hpp"
 
-address ProgrammableUpcallHandler::generate_optimized_upcall_stub(jobject receiver, Method* entry,
-                                                                  BasicType* in_sig_bt, int total_in_args,
-                                                                  BasicType* out_sig_bt, int total_out_args,
-                                                                  BasicType ret_type,
-                                                                  jobject jabi, jobject jconv,
-                                                                  bool needs_return_buffer, int ret_buf_size) {
+address UpcallLinker::generate_optimized_upcall_stub(jobject receiver, Method* entry,
```
Comment (mcimadamore) [inline src/hotspot/cpu/riscv/upcallLinker_riscv.cpp:30]: Hasn't "generated_optimized_upcall_stub" changed too?

**Review by mcimadamore — COMMENTED** [review]
Looks good - I spotted a potential issue in the risc-v port

File: src/hotspot/cpu/riscv/upcallLinker_riscv.cpp (hunk 2 above)
Comment (JornVernee) [reply src/hotspot/cpu/riscv/upcallLinker_riscv.cpp:30]: Right, I missed that one. Good catch!

**Comment by JornVernee** [comment]
Addressed now. I also noticed some names in strings that weren't updated yet. Updated those as well.

**Review by mcimadamore — APPROVED** [review]

**Review by robehn — APPROVED** [review]
Note that some find downcall/upcall unintuitive.
We may need revisit this.

Thanks

**Pull request closed without merging**